raiden/network/transport/matrix/transport.py

```diff
@@ -1082,52 +1188,82 @@ def _send_with_retry(self, queue: MessagesQueue) -> None:
         retrier = self._get_retrier(queue.queue_identifier.recipient)
         retrier.enqueue(queue_identifier=queue.queue_identifier, messages=queue.messages)
 
-    def _multicast_raw(
+    def _multicast_services(
         self,
         data: str,
         device_id: str = "*",
     ) -> None:
         assert self._raiden_service is not None, "_raiden_service not set"
 
-        # Construct the user without the help of the `UserAddressManager`
-        # This will not check for presence of the user-id and will
-        # ONLY send the data to the user-ids of the services on the node's connected homeserver.
-        user_ids = list()
-        for address in self.services_addresses.keys():
-            user_ids.append(f"@{to_normalized_address(address)}:{self._server_name}")
+        user_ids = {make_user_id(addr, self._server_name) for addr in self.services_addresses}
+        return self._send_to_device_raw(user_ids, data, device_id)
+
+    def _send_to_device_raw(
+        self,
+        user_ids: Set[UserID],
+        data: str,
+        device_id: str = "*",
+        message_type: MatrixMessageType = MatrixMessageType.TEXT,
+    ) -> None:
+        # Sends data to multiple users via to-device in a single Matrix API call
+        assert self._raiden_service is not None, "_raiden_service not set"
 
         messages = {
-            user_id: {device_id: {"msgtype": MatrixMessageType.TEXT.value, "body": data}}
+            user_id: {device_id: {"msgtype": message_type.value, "body": data}}
             for user_id in user_ids
         }
 
         self.log.debug(
-            "Multicast (`to-device`)",
+            "Send to-device message",
             device_id=device_id,
             receivers=user_ids,
+            multicast=bool(len(user_ids) > 1),
             data=data.replace("\n", "\\n"),
         )
 
         self._client.api.send_to_device(event_type="m.room.message", messages=messages)
 
+    def _get_possible_user_ids(
+        self, address: Address, address_metadata: AddressMetadata = None
+    ) -> Set[UserID]:
+        """Construct possible user-ids from the address
+
+        This will take the information from the connected matrix homeservers and
+        an optional AddressMetadata dictionary.
+        If the address metadata doesn't further specify what user-id to use for that
+        address, then this method will simply construct all possible user-ids on all
+        known matrix homeservers.
+        """
+        user_ids = set()
+        if address_metadata is not None:
+            user_id = address_metadata.get("user_id")
+            if is_valid_userid(user_id):
+                user_id = cast(UserID, user_id)
+                user_ids.add(user_id)
+        if not user_ids:
+            user_ids = {
+                make_user_id(address, server_name) for server_name in self._all_server_names
+            }
+        return user_ids
```

Comment:
I'm still not a fan of this approach, but a PR isn't the right place to discuss it.